cg/services/file_delivery/file_formatter_service/delivery_file_formatter.py

```diff
@@ -0,0 +1,56 @@
+import os
+from pathlib import Path
+from cg.constants.delivery import INBOX_NAME
+from cg.services.file_delivery.fetch_file_service.models import DeliveryFiles
+from cg.services.file_delivery.file_formatter_service.utils.case_file_formatter import (
+    CaseFileFormatter,
+)
+from cg.services.file_delivery.file_formatter_service.delivery_file_formatting_service import (
+    DeliveryFileFormattingService,
+)
+from cg.services.file_delivery.file_formatter_service.models import FormattedFiles, FormattedFile
+from cg.services.file_delivery.file_formatter_service.utils.sample_file_concatenation_formatter import (
+    SampleFileConcatenationFormatter,
+)
+from cg.services.file_delivery.file_formatter_service.utils.sample_file_formatter import (
+    SampleFileFormatter,
+)
+from cg.services.file_delivery.file_formatter_service.utils.utils import (
+    get_ticket_dir_path,
+    create_ticket_dir,
+)
+
+
+class DeliveryFileFormatter(DeliveryFileFormattingService):
+    """
+    Format the files to be delivered in the generic format.
+    Expected structure:
+    <customer>/inbox/<ticket_id>/<case_name>/<case_files>
+    <customer>/inbox/<ticket_id>/<sample_name>/<sample_files>
+    """
+
+    def __init__(
+        self,
+        case_file_formatter: CaseFileFormatter,
+        sample_file_formatter: SampleFileFormatter | SampleFileConcatenationFormatter,
+    ):
+        self.case_file_formatter = case_file_formatter
+        self.sample_file_formatter = sample_file_formatter
+
+    def format_files(self, delivery_files: DeliveryFiles) -> FormattedFiles:
+        """Format the files to be delivered and return the formatted files in the generic format."""
+        ticket_dir_path: Path = get_ticket_dir_path(delivery_files.sample_files[0].file_path)
+        create_ticket_dir(ticket_dir_path)
+        formatted_files: list[FormattedFile] = []
+        formatted_sample_files: list[FormattedFile] = self.sample_file_formatter.format_files(
+            moved_files=delivery_files.sample_files,
+            ticket_dir_path=get_ticket_dir_path(delivery_files.sample_files[0].file_path),
+        )
+        formatted_files.extend(formatted_sample_files)
+        if delivery_files.case_files:
```

Comment:
Could we give the DeliveryFiles class a default value for case_files? Then we could set it to an empty list

Reply:
I don't where what that would solve. Why is the if statement an issue?

Reply:
Every logic-statement in a function increases code-complexity so if possible It would be nice to have a function with as few indentations as possible. What does the extend function complain about? I tried and it seems to behave as expected with `[]` as input

Reply:
The case formatter also creates a directory that would fail without the expected input of a list of case_files. 
the case_file_formatter should format the given input and return a list of formatted files, which cannot be fulfilled with an empty list. 

not having the logic in its current form will create ambiguity elsewhere.